BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -0,0 +1,370 @@
+    controller.importAccount(name, privateKey: privateKey) { success, address in
+      self.updateKeyringInfo()
+      completion?(success, address)
+    }
+  }
+  
+  func addSecondaryAccount(
+    _ name: String,
+    json: String,
+    password: String,
+    completion: ((Bool, String) -> Void)? = nil
+  ) {
+    controller.importAccount(fromJson: name, password: password, json: json) { success, address in
+      completion?(success, address)
+    }
+  }
+  
+  func removeSecondaryAccount(forAddress address: String, completion: ((Bool) -> Void)? = nil) {
+    controller.removeImportedAccount(address) { success in
+      self.updateKeyringInfo()
+      completion?(success)
+    }
+  }
+  
+  func renameAccount(_ account: BraveWallet.AccountInfo, name: String, completion: ((Bool) -> Void)? = nil) {
+    let handler: (Bool) -> Void = { success in
+      self.updateKeyringInfo()
+      completion?(success)
+    }
+    if account.isImported {
+      controller.setDefaultKeyringImportedAccountName(account.address, name: name, completion: handler)
+    } else {
+      controller.setDefaultKeyringDerivedAccountName(account.address, name: name, completion: handler)
+    }
+  }
+  
+  func reset() {
+    controller.reset()
+    isOnboardingVisible = true
+    Self.resetKeychainStoredPassword()
+  }
+  
+  func privateKey(for account: BraveWallet.AccountInfo, completion: @escaping (String?) -> Void) {
+    if account.isPrimary {
+      controller.privateKey(forDefaultKeyringAccount: account.address) { success, key in
+        completion(success ? key : nil)
+      }
+    } else {
+      controller.privateKey(forImportedAccount: account.address) { success, key in
+        completion(success ? key : nil)
+      }
+    }
+  }
+  
+  func notifyUserInteraction() {
+    if keyring.isLocked {
+      // Auto-lock isn't running until the keyring is unlocked
+      return
+    }
+    controller.notifyUserInteraction()
+  }
+  
+  // MARK: - Keychain
+  
+  private static let passwordKeychainKey = "brave-wallet-password"
+  
+  /// Stores the users wallet password in the keychain so that they may unlock using biometrics/passcode
+  static func storePasswordInKeychain(_ password: String) -> Bool {
+    guard let passwordData = password.data(using: .utf8) else { return false }
+    #if targetEnvironment(simulator)
+    // There is a bug with iOS 15 simulators when attempting to add a keychain item with
+    // `kSecAttrAccessControl` set. This of course means that on simulator we will not ask for biometrics
+    // and it will just auto-fill the password field but at least to set it up you still need to enable
+    // biometrics on the simulator
+    //
+    // Last checked: Xcode 13.1 (13A1030d)
+    let query: [String: Any] = [
+      kSecClass as String: kSecClassGenericPassword,
+      kSecAttrAccount as String: passwordKeychainKey,
+      kSecValueData as String: passwordData
+    ]
+    #else
+    let accessControl = SecAccessControlCreateWithFlags(
```

Comment:
is this keychain entry available to other apps or just brave?

Reply:
Only Brave has access to keychain items added inside the app